src/scenarios.py

```diff
@@ -0,0 +1,148 @@
+import argparse
+import csv
+import yaml
+
+import numpy as np
+from itertools import product
+
+from sim import simulate as sim
+
+#TODO add charger_density once supported in sim.py
+#TODO include support for depot_chargers. Not sure the best way to scale this
+# to the number of actual chargers that would be present at a depot.
+def genScenarios(battery_cap_kwh,nondepot_charger_rate,):
+    '''Generates a list of tuples of scenarios
+        args: all tuples of a start (inclusive), end (exclusive), and step 
+        size values. i.e. battery_cap_kwh=(125,700,100) 
+    '''
+    #TODO include error handling
+    bat_range = np.arange(*battery_cap_kwh)
+    non_depot_range = np.arange(*nondepot_charger_rate)
+    scen_opts = [bat_range,non_depot_range,]
+    scens = list(product(*scen_opts))
+    keys = list(range(len(scens)))
+    scens = dict(zip(keys, scens))
```

Comment:
Don't need explicit lists here since dict consumes the generators

Reply:
scens needs to be a list so we can get the length for building the range of keys required for dict. Unless you know a more elegant way of doing that.